### PR TITLE
LPS-121262 Handle hyphens in topics names

### PR DESCRIPTION
**Pull request by LuismiBarcos**

### CODE REVIEW

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/utils/utils.es.js [hunk 1]
```diff
@@ -122,15 +122,26 @@ export function normalize(ratingValue) {
 }
 
 export function stringToSlug(text) {
+	const hyphens = /-+/g;
 	const whiteSpaces = /\s+/g;
 
-	return text.replace(whiteSpaces, '-').toLowerCase();
+	text = text.toLowerCase();
+	text = text.replace(hyphens, '--');
+
```
Comment (javierdearcos) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/utils/utils.es.js:130]: I've just realized that this is already ordered alphabetically... (hypens and whiteSpaces). So maybe Brian was referring to add an breakline between the toLowerCase method and the first replace method? I think that you were right when we talked about it and the breakline is needed between this to lines

File: modules/apps/questions/questions-web/test/js/utils/utils.es.js [hunk 2]
```diff
@@ -21,5 +21,34 @@ describe('utils', () => {
 				'lorem ipsum dolor'
 			);
 		});
+
+		it('return a text with a hyphen and a space', () => {
+			expect(Utils.slugToText('Lorem--ipsum-dolor')).toEqual(
+				'lorem-ipsum dolor'
+			);
+			expect(Utils.slugToText('Lorem--ipsum--dolor')).toEqual(
```
Comment (javierdearcos) [inline modules/apps/questions/questions-web/test/js/utils/utils.es.js:29]: make a new it block or replace the text to be more generic and fit all the cases that are beeing tested

File: modules/apps/questions/questions-web/test/js/utils/utils.es.js [hunk 3]
```diff
@@ -21,5 +21,34 @@ describe('utils', () => {
...
+			);
+		});
+	});
+
+	describe('stringToSlug', () => {
+		it('return a text with hyphens instead of spaces', () => {
+			expect(Utils.stringToSlug('Lorem ipsum dolor')).toEqual(
+				'lorem-ipsum-dolor'
+			);
+			expect(Utils.stringToSlug('Lorem  ipsum  dolor')).toEqual(
```
Comment (javierdearcos) [inline modules/apps/questions/questions-web/test/js/utils/utils.es.js:40]: We add the trim functionality?

File: modules/apps/questions/questions-web/test/js/utils/utils.es.js [hunk 4]
```diff
@@ -21,5 +21,34 @@ describe('utils', () => {
...
+				'lorem-ipsum dolor'
+			);
+			expect(Utils.slugToText('Lorem--ipsum--dolor')).toEqual(
+				'lorem-ipsum-dolor'
+			);
+		});
+	});
+
+	describe('stringToSlug', () => {
+		it('return a text with hyphens instead of spaces', () => {
```
Comment (javierdearcos) [inline modules/apps/questions/questions-web/test/js/utils/utils.es.js:36]: the same as above

**Review by javierdearcos — COMMENTED** [review]
I've put some things to review

File: modules/apps/questions/questions-web/test/js/utils/utils.es.js (hunk 3 above)
Comment (LuismiBarcos) [reply modules/apps/questions/questions-web/test/js/utils/utils.es.js:40]: Nop. Right now, what we do is replace the white spaces ('\s') for a hyphen, it doesn't matter how many white spaces contain the text. So, it is not totally a trim.

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Pull request closed without merging**